### ISSUE TITLE
[java] DD False Positive in DataflowAnomalyAnalysis

### ISSUE THREAD

**Issue by SukJinKim** [pmd/pmd#1749]
PMD Version: 6.9.0

Rule: [DataflowAnomalyAnalysis](https://pmd.github.io/latest/pmd_rules_java_errorprone.html#dataflowanomalyanalysis)

Description: To ensure that I understand the notion of data flow correctly, I tested with simple codes.  I found something weird during the test. I intuitively thought that the data flow for variable a in the code below is DRD. However, DD anomaly occurred for variable a.
I suspect that the rule do not read the code according to the actual flow of variables, but read it in the order of the code(from left to right). Is is intended or fault for the rule design?

```
public void Test(){
int a = 0;
a = a + 3;
}
```

**Comment by jsotuyod** [comment]
@SukJinKim thanks for the report.

I fear that the Dataflow codebase is not up to the same standards as the rest of PMD. It's still unaware of too many (not so new) Java constructs (ie: foreach, lambdas, switch statements, etc.); so I'd not trust it so much.

What you describe is effectively a false positive (strictly speaking, initializing `a` to 0 instead of 3 directly is a waste of CPU cycles, but that rule isn't trying to be smart about it).
